src/node_crypto.cc

```diff
@@ -5421,11 +5421,15 @@ void PBKDF2(const FunctionCallbackInfo<Value>& args) {
 // Only instantiate within a valid HandleScope.
 class RandomBytesRequest : public AsyncWrap {
  public:
-  RandomBytesRequest(Environment* env, Local<Object> object, size_t size)
+  RandomBytesRequest(Environment* env,
+                     Local<Object> object,
+                     size_t size,
+                     size_t offset)
       : AsyncWrap(env, object, AsyncWrap::PROVIDER_CRYPTO),
         error_(0),
         size_(size),
-        data_(node::Malloc(size)) {
+        data_(node::Malloc(size)),
```

Comment:
This looks like a memory leak to me when `randomFill` is used – maybe it would be better to pass the memory behind the input `Buffer` as a parameter to `RandomBytesRequest`, and store that in `data_`?

That might allow you to get rid of the extra `offset` parameter and the memory copying in `RandomBytesCheck`, too.